sdk/core/Azure.Core/src/Shared/CallerShouldAuditAttribute.cs

```diff
@@ -18,6 +18,23 @@ namespace Azure.Core
     [AttributeUsage(AttributeTargets.Method)]
     internal class CallerShouldAuditAttribute : Attribute
     {
+        /// <summary>
+        /// Creates a new instance of <see cref="CallerShouldAuditAttribute"/>.
+        /// </summary>
+        public CallerShouldAuditAttribute()
+            : this(reason: null)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="CallerShouldAuditAttribute"/>.
+        /// </summary>
+        /// <param name="reason"> Sets a description or link to the rationale for potentially auditing this operation. </param>
+        public CallerShouldAuditAttribute(string? reason)
```

Comment:
Why is this nullable?  Anyone who didn't want to pass it should just call the other .ctor.

Reply:
it's defined as 'string?' so that the other ctor could just calling it. The main purpose is to keep one ctor do the implementation while others just call that one when possible and makes sense, which would be easier to maintain in the future. i.e. you just need to focus on one ctor when updating the implementation instead of needing to update multiple ctor and having the risk to miss one. thanks.